foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiChatCompletionPrompt.java

```diff
@@ -0,0 +1,80 @@
+package com.sap.ai.sdk.foundationmodels.openai;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.annotation.Nonnull;
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+/** Represents a prompt that contains one or more messages for OpenAI chat completion. */
+@Accessors(fluent = true)
+@Data
+public class OpenAiChatCompletionPrompt {
+  @Nonnull private final List<OpenAiMessage> messages = new ArrayList<>();
+
+  @Nonnull private final List<String> stop = new ArrayList<>();
+
+  private OpenAiChatCompletionPrompt(@Nonnull final String message) {
+    this.messages.add(OpenAiMessage.user(message));
+  }
+
+  private OpenAiChatCompletionPrompt(@Nonnull final List<OpenAiMessage> messages) {
+    this.messages.addAll(messages);
+  }
+
+  /**
+   * Creates an OpenAiChatCompletionPrompt with a single message.
+   *
+   * @param message the message to be added to the prompt
+   * @return a new OpenAiChatCompletionPrompt instance
+   */
+  @Nonnull
+  public static OpenAiChatCompletionPrompt create(@Nonnull final String message) {
+    return new OpenAiChatCompletionPrompt(message);
```

Comment:
(Minor)

Please consider method chaining and minimal constructor:

```suggestion
@RequiredArgsConstructor(access = lombok.AccessLevel.PRIVATE)
public class OpenAiChatCompletionPrompt {
  @Nonnull private final List<OpenAiMessage> messages;

  @Nonnull private final List<String> stop = new ArrayList<>();

  /**
   * Creates an OpenAiChatCompletionPrompt with a single message.
   *
   * @param message the message to be added to the prompt
   * @return a new OpenAiChatCompletionPrompt instance
   */
  @Nonnull
  public static OpenAiChatCompletionPrompt create(@Nonnull final String message) {
    return create(OpenAiMessage.user(message));
```